### DISCUSSION TITLE
Show modified time of the file under the cursor on the status line

### DISCUSSION THREAD

**Discussion by GeordyJ** [sxyazi/yazi#1658] [2. Show and tell]
I just wanted to share how I was able to add the modified time of the file under the cursor in the status line. You just need to add the following code to the `yazi/init.lua` file. I prefer this rather than using the linemodes.

```lua
Status:children_add(function()
	local h = cx.active.current.hovered
	return ui.Line({
		ui.Span(os.date(_, tostring(h.cha.mtime):sub(1, 10))):fg("blue"),
		ui.Span(" "),
	})
end, 500, Status.RIGHT)
```
<img width="1470" alt="Screenshot 2024-09-18 at 7 34 12 PM" src="https://github.com/user-attachments/assets/57002d5f-cab7-4c34-942e-e2d9b43e07f2">

You can modify it to see other properties, see the [docs](https://yazi-rs.github.io/docs/plugins/types#shared.cha)

**Comment by nighthawk469** [comment]
I would like if this is on by default. Same as `lf`

**Reply by GeordyJ** [reply]
No it is not on by default, taxi does not show the modified time in the red box in the pic like lf.

**Reply by nighthawk469** [reply]
Also I get this error when I use this script
![image](https://github.com/user-attachments/assets/b4e18476-ac1b-4b28-ac19-30b65eb837dc)

**Reply by nighthawk469** [reply]
do you mean yazi not taxi ?

**Reply by GeordyJ** [reply]
Sorry for the typo it’s yazi (commenting from phone).
Secondly, to get rid of the warning replace ‘cha.modified’ with ‘cha.mtime’

**Reply by nighthawk469** [reply]
Thanks, works good.

**Comment by pavvd** [comment]
Thank you, I was looking for something like this. Here is modified variant, displaying both B and Mtime in different format:
```lua
Status:children_add(function()
    local h = cx.active.current.hovered
    local elements = {}

    local btime_formatted = nil
    if h and h.cha and h.cha.btime then
        local timestamp_num = tonumber(h.cha.btime)
        if timestamp_num and timestamp_num > 0 then
            btime_formatted = os.date("%Y-%m-%d %H:%M", math.floor(timestamp_num))
        end
    end

    local mtime_formatted = nil
    if h and h.cha and h.cha.mtime then
        local timestamp_num = tonumber(h.cha.mtime)
        if timestamp_num and timestamp_num > 0 then
            mtime_formatted = os.date("%Y-%m-%d %H:%M", math.floor(timestamp_num))
        end
    end

    if btime_formatted then
        table.insert(elements, ui.Span("B:"):fg("cyan"))
        table.insert(elements, ui.Span(btime_formatted.. " "):fg("blue"))
    end

    if mtime_formatted then
        table.insert(elements, ui.Span("M:"):fg("cyan"))
        table.insert(elements, ui.Span(mtime_formatted.. " "):fg("blue"))
    end

    return ui.Line(elements)
end, 500, Status.RIGHT)
```
